x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_panel.tsx

```diff
@@ -134,6 +134,10 @@ export const PackagePolicyInputPanel: React.FunctionComponent<{
         (ds) => ds.dataset === packagePolicyInputStream?.data_stream.dataset
       );
 
+      // It seems this logic does check for the data stream setting. What if template and data stream are not in sync (no rollover yet)
+      // Is there a way in Elasticsearch to say: For data stream X, what template will apply? The reason I'm asking
+      // is that there is a chance that users made some modification in the settings or that for a namespace an additional
+      // template was set in place. So if we only look at the data stream, the info might be wrong / outdated.
```

Comment:
> It seems this logic does check for the data stream setting. What if template and data stream are not in sync (no rollover yet)

This is correct. Our current implementation does not consider whether the currently "visible" value of a given toggle has actually been applied (rolled over) to the given data stream. I think this okay-ish for now, since the features are presented quite heavily as experimental, and I expect them to be removed in favor of package spec support as the source of truth in the near future. 

If we had automated rollover https://github.com/elastic/kibana/issues/143448, I think this issue would go away, as well. So, if that's low hanging fruit as I think it is, maybe that's the quick solve here. 

> Is there a way in Elasticsearch to say: For data stream X, what template will apply?
> 
> The reason I'm asking is that there is a chance that users made some modification in the settings or that for a namespace an additional template was set in place. So if we only look at the data stream, the info might be wrong / outdated.

This I do not know. Poking around the index template and data stream docs for Elasticserach didn't turn up much, but there might be some internal API to expose this. Or maybe there's a clever search operation we could use here. Would be interesting to look into it further, but not something I'm immediately aware of.

Reply:
Let me know in case the Fleet team is interested in having such an API, I might be able to contribute it to Elasticsearch.